restic/cfg/config.go

```diff
@@ -0,0 +1,91 @@
+package cfg
+
+import (
+	"fmt"
+	"strings"
+)
+
+const (
+	// RestoreTypeS3 indicates that the restore shall be done to an S3 endpoint.
+	RestoreTypeS3 = "s3"
+
+	// RestoreTypeFolder indicates that the restore shall be done to a folder,
+	// usually a RWX PVC mounted to the Pod of the restore process.
+	RestoreTypeFolder = "folder"
+)
+
+var (
+	// Config contains the values of the user-provided configuration of the operator module,
+	// combined with the default values as defined in operator.Command.
+	Config = &Configuration{}
+)
+
+// Configuration holds a strongly-typed tree of the configuration
+type Configuration struct {
+	DoCheck   bool
+	DoPrune   bool
+	DoRestore bool
+	DoArchive bool
+
+	BackupCommandAnnotation       string
+	BackupFileExtensionAnnotation string
+	BackupDir                     string
+
+	PromURL    string
+	WebhookURL string
+
+	Hostname   string
+	KubeConfig string
+
+	ResticBin        string
+	ResticRepository string
+	ResticOptions    string
+
+	RestoreDir         string
+	RestoreS3Endpoint  string
+	RestoreS3AccessKey string
+	RestoreS3SecretKey string
+	RestoreSnap        string
+	RestoreType        string
+	RestoreFilter      string
+	VerifyRestore      bool
+	RestoreTrimPath    bool
+
+	PruneKeepLast    *int
+	PruneKeepHourly  *int
+	PruneKeepDaily   *int
+	PruneKeepWeekly  *int
+	PruneKeepMonthly *int
+	PruneKeepYearly  *int
```

Comment:
Q: Does urfave/cli correctly set these values for pointers?

Reply:
You raise a good point here and I will have to check whether these work as I expect.

Reply:
You were right. The values were not handled as I expected. I've updated the code to handle them.